src/Components/Components/src/Auth/AuthorizeRouteView.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Combines the behaviors of <see cref="AuthorizeView"/> and <see cref="RouteView"/>,
+    /// so that it displays the page matching the specified route but only if the user
+    /// is authorized to see it.
+    ///
+    /// Additionally, this component supplies a cascading parameter of type <see cref="Task{AuthenticationState}"/>,
+    /// which makes the user's current authentication state available to descendants.
+    /// </summary>
+    public sealed class AuthorizeRouteView : RouteView
+    {
+        /// <summary>
+        /// The content that will be displayed if the user is not authorized.
+        /// </summary>
+        [Parameter] public RenderFragment<AuthenticationState> NotAuthorized { get; set; }
+
+        /// <summary>
+        /// The content that will be displayed while asynchronous authorization is in progress.
+        /// </summary>
+        [Parameter] public RenderFragment Authorizing { get; set; }
+
+        /// <inheritdoc />
+        protected override void Render(RenderTreeBuilder builder)
+        {
+            // TODO: Consider merging the behavior of CascadingAuthenticationState into this
+            // component (i.e., rendering the CascadingValue directly) to avoid the extra
+            // layer of component nesting and eliminate CascadingAuthenticationState as public API.
+            builder.OpenComponent<CascadingAuthenticationState>(0);
+            builder.AddAttribute(1, nameof(CascadingAuthenticationState.ChildContent), (RenderFragment)(builder =>
```

Comment:
If this is annoying, we could add back `RenderFragment` and `RenderFragment<T>` as overloads to the RTB. I think they were removed a while ago for ambiguity reasons, but it's probably not an issue anymore.

Reply:
I ended up removing this in a later commit, but I agree there would be cases where an overload for `RenderFragment<T>` would save an allocation, as it would eliminate the closure.

No urgency on this - it's something we could add later.